code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -0,0 +1,159 @@
+			continue
+		light.turn_light(null, FALSE, duration, TRUE, TRUE)
+	succeed_activate()
+	add_cooldown()
+
+
+// ***************************************
+// *********** Gravity Crush
+// ***************************************
+#define ADD_FILTER (1<<0)
+#define REMOVE_FILTER (1<<1)
+#define GRAV_CRUSH (1<<2)
+
+#define WINDUP_GRAV 2 SECONDS
+
+/datum/action/xeno_action/activable/gravity_crush
+	name = "Gravity Crush"
+	action_icon_state = "fortify"
+	mechanics_text = "Increases the localized gravity in an area and crushes structures."
+	ability_name = "Gravity crush"
+	plasma_cost = 100
+	cooldown_timer = 30 SECONDS
+	keybind_signal = COMSIG_XENOABILITY_GRAVITY_CRUSH
+	/// How far can we use gravity crush
+	var/king_crush_dist = 5
+
+/datum/action/xeno_action/activable/gravity_crush/on_cooldown_finish()
+	to_chat(owner, "<span class='warning'>Our psychic aura restores itself. We are ready to gravity crush again.</span>")
+	return ..()
+
+/datum/action/xeno_action/activable/gravity_crush/can_use_ability(atom/A, silent, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(!owner.line_of_sight(A, king_crush_dist))
+		if(!silent)
+			to_chat(owner, "<span class='warning'>We must get closer to crush, our mind cannot reach this far.</span>")
+		return FALSE
+
+/datum/action/xeno_action/activable/gravity_crush/use_ability(atom/A)
+	var/list/turfs = RANGE_TURFS(1, A)
+	playsound(A, 'sound/effects/bomb_fall.ogg', 75, FALSE)
+	act_on_turfs(ADD_FILTER, turfs)
+	if(!do_after(owner, WINDUP_GRAV, FALSE, owner, BUSY_ICON_DANGER))
+		act_on_turfs(REMOVE_FILTER, turfs)
+		return fail_activate()
+	act_on_turfs(REMOVE_FILTER|GRAV_CRUSH, turfs)
+	succeed_activate()
+	add_cooldown()
+	A.visible_message("<span class='warning'>[A] collapses inward as its gravity suddenly increases!</span>")
+
+///Will do the selected actions on all turfs and items in the turf in the list of turfs
+/datum/action/xeno_action/activable/gravity_crush/proc/act_on_turfs(action_flags, list/turfs)
+	for(var/turf/targetted AS in turfs)
+		if(action_flags & ADD_FILTER)
+			targetted.add_filter("crushblur", 1, radial_blur_filter(0.3))
+		if(action_flags & REMOVE_FILTER)
+			targetted.remove_filter("crushblur")
+		for(var/atom/movable/item AS in targetted.contents)
+			if(action_flags & ADD_FILTER)
+				item.add_filter("crushblur", 1, radial_blur_filter(0.3))
+				addtimer(CALLBACK(item, /atom.proc/remove_filter, "crushblur"), WINDUP_GRAV) //Fail safe if the item moves out of the zone.
+			if(action_flags & REMOVE_FILTER)
+				item.remove_filter("crushblur")
+			if(action_flags & GRAV_CRUSH)
+				if(isliving(item))
+					var/mob/living/mob_crushed = item
+					if(mob_crushed.stat == DEAD)//No abuse of that mechanic for some permadeath
+						continue
+					if(mob_crushed.faction == FACTION_XENO) //Don't harm the xeno. I thought it was better than a istype, could be wrong
+						continue
+				item.ex_act(EXPLODE_HEAVY)	//crushing without damaging the nearby area
```

Comment:
why does this add AND remove filters AND handle crushing

Reply:
You said no copy paste?
I want to remove the filter AND crush as the same time